.github/workflows/build.yaml

```diff
@@ -70,15 +72,13 @@ jobs:
     - name: Install dependencies
       run: |
         python -m pip install --upgrade pip
-        pip install setuptools wheel twine
+        python -m pip install poetry
     - name: Build package and publish to PyPI
-      env:
-        TWINE_USERNAME: __token__
-        TWINE_PASSWORD: ${{ secrets.PYPI_OBS_PORTAL_API_TOKEN }}
       run: |
-        python setup.py sdist bdist_wheel
-        twine upload dist/*
-      
+        poetry build -f wheel
+        poetry build -f sdist
+        poetry publish -u "__token__" -p "${{ secrets.PYPI_OBS_PORTAL_API_TOKEN }}"
```

Comment:
Yes, these might contain special chars (that bash might not like), so generally a good idea to quote CLI arguments. Actually, I'm going to change this to single quotes to be even more cautious.

And the reason I moved these from environment variables to template, is because I believe that GitHub actions doesn't print out templated secrets in the logs. I don't know what the behavior is for environment variables.